uhabits-android/src/main/java/org/isoron/uhabits/activities/habits/list/views/HabitCardView.kt

```diff
@@ -115,6 +115,13 @@ class HabitCardView(
             numberPanel.threshold = value
         }
 
+    var notes
```

Comment:
I don't pretend to understand your code, but I find it surprising that there is a getter that returns one variable and a setter that sets two of them. Do these variables ever differ?

Reply:
They are actually the same boolean array indicating whether there is notes or not. The habit `values` are also set the same way.

Reply:
Why can't we then remove one of these variables?

Reply:
when `checkmarkPanel.notes` is set, it sets `button.hasNotes` (`CheckMarkButtonView`) which is used to check whether there is any notes and mark the button with a small circle at the top right corner. If only `checkmarkPanel` value is set then we will not be able to check whether number type habit has any notes or not and might not be able to mark the button with a circle and vice versa.